src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/TargetFrameworkMoniker.cs

```diff
@@ -2,25 +2,200 @@
+        public static readonly TargetFrameworkMoniker NetCoreApp21 = new(NetCoreApp, new Version(2, 1));
+        public static readonly TargetFrameworkMoniker NetCoreApp30 = NetCoreApp21 with { FrameworkVersion = new Version(3, 0) };
+        public static readonly TargetFrameworkMoniker NetCoreApp31 = NetCoreApp30 with { FrameworkVersion = new Version(3, 1) };
+        public static readonly TargetFrameworkMoniker Net45 = new(Net, new Version(4, 5));
+        public static readonly TargetFrameworkMoniker Net46 = Net45 with { FrameworkVersion = new Version(4, 6) };
+        public static readonly TargetFrameworkMoniker Net461 = Net45 with { FrameworkVersion = new Version(4, 6, 1) };
+        public static readonly TargetFrameworkMoniker Net462 = Net45 with { FrameworkVersion = new Version(4, 6, 2) };
+        public static readonly TargetFrameworkMoniker Net47 = Net45 with { FrameworkVersion = new Version(4, 7) };
+        public static readonly TargetFrameworkMoniker Net471 = Net45 with { FrameworkVersion = new Version(4, 7, 1) };
+        public static readonly TargetFrameworkMoniker Net472 = Net45 with { FrameworkVersion = new Version(4, 7, 2) };
+        public static readonly TargetFrameworkMoniker Net48 = Net45 with { FrameworkVersion = new Version(4, 8) };
+        public static readonly TargetFrameworkMoniker Net50 = NetCoreApp30 with { Framework = Net, FrameworkVersion = new Version(5, 0) };
+        public static readonly TargetFrameworkMoniker Net50_Windows = Net50 with { Platform = Platforms.Windows };
+        public static readonly TargetFrameworkMoniker Net50_Linux = Net50 with { Platform = Platforms.Linux };
+        public static readonly TargetFrameworkMoniker Net60 = Net50 with { FrameworkVersion = new Version(6, 0) };
+        public static readonly TargetFrameworkMoniker Net60_Linux = Net60 with { Platform = Platforms.Linux };
+        public static readonly TargetFrameworkMoniker Net60_Windows = Net60 with { Platform = Platforms.Windows };
+#pragma warning restore CA1707 // Identifiers should not contain underscores
 
-        private const string NetPrefix = "net";
+#pragma warning disable CA1034 // Nested types should not be visible
+        public static class Platforms
+        {
+            public const string Windows = "windows";
+            public const string Linux = "linux";
+        }
+#pragma warning restore CA1034 // Nested types should not be visible
 
-        public override string ToString() => Name;
+        private readonly string? _platform;
+        private readonly Version? _platformVersion;
+        private readonly string _framework;
+        private readonly Version _frameworkVersion;
 
-        public bool IsFramework => !IsNetStandard && !IsNetCore;
+        public TargetFrameworkMoniker(string framework, Version frameworkVersion)
+        {
+            _framework = framework;
```

Comment:
throw an exception if **framework** is not one of the valid frameworks?

this would prevent code that inadvertently sets the platform as part of the framework.

Reply:
We just end up ignoring it in that case. I've added some tests for that now.